### PR TITLE
feature/cplp-1834-document-types

### PR DESCRIPTION
**Pull request by qxz2mqe**
https://jira.catena-x.net/browse/CPLP-1834

### CODE REVIEW

File: src/portalbackend/PortalBackend.PortalEntities/Enums/DocumentTypeId.cs [hunk 1]
```diff
@@ -25,10 +25,12 @@ public enum DocumentTypeId
     CX_FRAME_CONTRACT = 1,
     COMMERCIAL_REGISTER_EXTRACT = 2,
     APP_CONTRACT = 3,
-    APP_DATA_DETAILS = 4,
-    ADDITIONAL_DETAILS = 5,
-    APP_LEADIMAGE = 6,
-    APP_IMAGE = 7,
-    SELF_DESCRIPTION = 8,
-    APP_TECHNICAL_INFORMATION = 9
+    ADDITIONAL_DETAILS = 4,
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.PortalEntities/Enums/DocumentTypeId.cs:28]: you cannot just rearrange the enum-values without migrating existing data. Before the change ADDITIONAL_DETAILS was 5, now it is 4, APP_LEAD_IMAGE was 6, now it is 5 etc...
existing records in db refer to enum-values 1,3,4,5,6,7, that means documents that were marked as APP_DATA_DETAILS now seem to be ADDITIONAL_DETAILS, ADDITIONAL_DETAIL-documents now appear as APP_LEAD_IMAGES.
Either the migration must contain code that modifies those entries appropriately (both in up and down) or the value of the deleted enum-member APP_DATA_DETAILS is just skipped while the new members are just added after the old ones (there wouldn't be a member 4 then...). In any case the migration must contain sql-code that removes all obsolete document-data that refers to the to-be-deleted member APP_DATA_DETAILS.

**Review by ntruchsess — CHANGES_REQUESTED** [review]

File: src/portalbackend/PortalBackend.PortalEntities/Enums/DocumentTypeId.cs (hunk 1 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.PortalEntities/Enums/DocumentTypeId.cs]: Resolved

File: src/portalbackend/PortalBackend.Migrations/Migrations/20230213100306_CPLP-1834-AddRemoveDocumentTypeS.cs [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+                values: new object[,]
+                {
+                    { 10, "CONFORMITY_APPROVAL_CONNECTOR" },
+                    { 11, "CONFORMITY_APPROVAL_BUSINESS_APPS" }
+                });
+        }
+
+        protected override void Down(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.DeleteData(
```
Comment (ntruchsess) [inline src/portalbackend/PortalBackend.Migrations/Migrations/20230213100306_CPLP-1834-AddRemoveDocumentTypeS.cs:54]: as 4 is being re-mapped from APP_DATA_DETAILS to CONFORMITY_APPROVAL_REGISTRATION when going back everything that has been uploaded as CONFORMITY_APPROVAL_REGISTRATION needs to be deleted pretty much the same way as APP_DATA_DETAILS-documents in up().

**Review by ntruchsess — CHANGES_REQUESTED** [review]

**Review by ntruchsess — CHANGES_REQUESTED** [review]
seeding-data contains documents where "document_type_id": 4 is being set. All related records (e.g. offer_assigned_documents) and affected unit-tests need to be adjusted

File: src/portalbackend/PortalBackend.Migrations/Migrations/20230213100306_CPLP-1834-AddRemoveDocumentTypeS.cs (hunk 2 above)
Comment (qxz2mqe) [reply src/portalbackend/PortalBackend.Migrations/Migrations/20230213100306_CPLP-1834-AddRemoveDocumentTypeS.cs:54]: fixed

**Review by ntruchsess — CHANGES_REQUESTED** [review]
As being stated before - seeding data needs to be adjusted as well (documents.consortia.json contains a document referencing document_type_id 4. That entry and anything that creates an fk-reference to it needs to be removed)

**Review by ntruchsess — APPROVED** [review]

**Pull request merged**